### DISCUSSION TITLE
Progress not being shown

### DISCUSSION THREAD

**Discussion by alvincrespo** [rubocop/rubocop#12920] [Q&A]
Hey all - I'm running into an issue where running Rubocop isn't showing it's progress until the very end so it seems stuck but it's actually running, just not showing the progress. Once it's done all the dots appear in the terminal at once. Curious if anyone else is experiencing this?

My system is:

```
ruby -v
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [arm64-darwin23]
which ruby
/Users/alvincrespo/.rbenv/shims/ruby
bundle exe rubocop --version
1.63.5
```

I'm using iTerm2 + oh_my_zsh (using the built in plugins `plugins=(git nvm docker rbenv)`).

**Comment by koic** [comment]
This behavior is due to the implementation of the default `--parallel` processing. For example, if you switch to `--no-parallel`, a progress character `.` is displayed immediately, but generally, it is slower than the default `--parallel` on multicore systems. The related issue is #10072.
